File: src/main/java/step1/Rental.java
package step1;

class Rental {
	private Movie movie;
	private int daysRented;

	public Rental(Movie movie, int daysRented) {
		this.movie = movie;
		this.daysRented = daysRented;
	}

	public int getDaysRented() {
		return daysRented;
	}

	public Movie getMovie() {
		return movie;
	}

	double getPrice() {
		return movie.getPrice(daysRented);
	}

	int getFrequentRenterPoints() {
		return movie.getFrequentRenterPoints(daysRented);
	}
}

File: src/main/java/step1/Rental.kt
package step1

class Rental(private val movie: Movie, val daysRented: Int) {

    // determine amounts for each line
    val price: Double
        get() {

            val daysRented = daysRented
            return movie.amountFor(daysRented)
        }

    // add frequent renter points
    val frequentRenterPoints: Int
        get() {
            val daysRented = daysRented

            return movie.getFrequentRenterPoints(daysRented)
        }

    val statementResult: String
        get() = "\t" + price.toString() + "(" + movie.title + ")" + "\n"
}